src/osd/OSD.cc

```diff
@@ -8864,7 +8987,11 @@ void OSD::enqueue_op(PG *pg, OpRequestRef& op)
 	   << " cost " << op->get_req()->get_cost()
 	   << " latency " << latency
 	   << " " << *(op->get_req()) << dendl;
-  pg->queue_op(op);
+  if (pg->can_handle_fast(op)) {
+    pg->queue_op_without_pg_lock(op);
```

Comment:
Well, this would be *very* wrong without extensive changes.  That code does require the pg lock right now.  Even in your design, the initial op was handled with the pg lock.  Furthermore, without *some* kind of lock, this would not even maintain op ordering on a single object.  What is the goal here?